core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -909,32 +918,35 @@ public boolean load() {
         }
     }
 
-    private boolean loadBaseGraphAndProperties() {
-        if (properties.loadExisting()) {
-            if (properties.containsVersion())
-                throw new IllegalStateException("The GraphHopper file format is not compatible with the data you are " +
-                        "trying to load. You either need to use an older version of GraphHopper or run a new import");
-            // check encoding for compatibility
-            String flagEncodersStr = properties.get("graph.flag_encoders");
+    private void loadEncodingManagerFromProperties(StorableProperties properties) {
+        if (properties.containsVersion())
+            throw new IllegalStateException("The GraphHopper file format is not compatible with the data you are " +
+                    "trying to load. You either need to use an older version of GraphHopper or run a new import");
+        String encodedValueStr = properties.get("graph.encoded_values");
+        ArrayNode evList = deserializeEncodedValueList(encodedValueStr);
+        List<EncodedValue> encodedValues = new ArrayList<>();
+        evList.forEach(serializedEV -> encodedValues.add(EncodedValueSerializer.deserializeEncodedValue(serializedEV.textValue())));
+
+        String flagEncodersStr = properties.get("graph.flag_encoders");
+        List<FlagEncoder> flagEncoders = Stream.of(flagEncodersStr.split(","))
+                .map(str -> flagEncoderFactory.deserializeFlagEncoder(str, name -> encodedValues.stream()
+                        .filter(ev -> ev.getName().equals(name))
+                        .findFirst()
+                        .orElseThrow(() -> new IllegalStateException("FlagEncoder " + str + " uses unknown encoded value: " + name))))
+                .collect(Collectors.toList());
 
-            if (!encodingManager.toFlagEncodersAsString().equalsIgnoreCase(flagEncodersStr)) {
-                throw new IllegalStateException("Flag encoders do not match:"
-                        + "\nGraphhopper config: " + encodingManager.toFlagEncodersAsString()
-                        + "\nLoaded graph:       " + flagEncodersStr
-                        + "\nChange configuration to match the graph or delete " + baseGraph.getDirectory().getLocation());
-            }
+        EncodingManager.Builder emBuilder = EncodingManager.start();
+        flagEncoders.forEach(emBuilder::addWithoutAddingEncodedValues);
+        encodedValues.forEach(emBuilder::add);
+        encodingManager = emBuilder.build();
```

Comment:
We deserialized the encoded values and flag encoders and now build the encoding manager. However, the flag encoders are handled a bit differently here: When we create the encoding manager from the config file the flag encoders are used to *create* certain encoded values (like car_access etc.), but here we already deserialized all the encoded values and need to skip the creation of encoded values when adding the flag encoders. Flag encoders will hopefully be removed soon anyway :)